libslide/src/bignum/compare.rs

```diff
@@ -0,0 +1,131 @@
+use crate::bignum::Bignum;
+use std::cmp::Ordering;
+
+impl Ord for Bignum {
+    fn cmp(&self, other: &Bignum) -> Ordering {
+        let greater: Ordering;
+        let less: Ordering;
+        if self.is_neg && !other.is_neg {
+            return Ordering::Less;
+        } else if other.is_neg && !self.is_neg {
+            return Ordering::Greater;
+        } else if other.is_neg && self.is_neg {
+            greater = Ordering::Less;
+            less = Ordering::Greater;
+        } else {
+            greater = Ordering::Greater;
+            less = Ordering::Less;
+        }
+        match self.int.len().cmp(&other.int.len()) {
+            Ordering::Greater => return greater,
+            Ordering::Less => return less,
+            _ => (),
+        }
+
+        for i in (0..other.int.len()).rev() {
+            match self.int[i].cmp(&other.int[i]) {
+                Ordering::Greater => return greater,
+                Ordering::Less => return less,
+                _ => (),
+            }
+        }
+
+        let smaller_dec_size: usize = if self.dec.len() > other.dec.len() {
+            other.dec.len()
+        } else {
+            self.dec.len()
+        };
+
+        for i in 0..smaller_dec_size {
+            match self.dec[i].cmp(&other.dec[i]) {
+                Ordering::Greater => return greater,
+                Ordering::Less => return less,
+                _ => (),
+            }
+        }
+
+        match other.dec.len().cmp(&self.dec.len()) {
+            Ordering::Equal => Ordering::Equal,
+            Ordering::Less => less,
+            Ordering::Greater => greater,
+        }
+    }
+}
+
+impl Eq for Bignum {}
+
+impl PartialEq for Bignum {
+    fn eq(&self, other: &Bignum) -> bool {
+        self.cmp(other) == Ordering::Equal
+    }
+}
+
+// returns true if the rhs is smaller then lhs
+impl PartialOrd for Bignum {
+    fn partial_cmp(&self, other: &Bignum) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    macro_rules! bignum_test_cmp {
+        ($($name: ident: $lhs:expr, $rhs:expr, $program: expr)*) => {
+        $(
+            #[test]
+            fn $name() {
+                use crate::bignum::Bignum;
+                let lhs = Bignum::new($lhs.to_string()).unwrap();
+                let rhs = Bignum::new($rhs.to_string()).unwrap();
+                match $program {
+                    "=" => assert!(lhs == rhs),
+                    "!=" => assert!(lhs != rhs),
+                    "<=" => assert!(lhs <= rhs),
+                    ">=" => assert!(lhs >= rhs),
+                    ">" => assert!(lhs > rhs),
+                    "<" => assert!(lhs < rhs),
+                    _ => panic!("Test input invalid"),
+                }
+            }
+        )*
+        }
+    }
+
+    mod cmp {
+        bignum_test_cmp! {
+            equal1: "1", "1", "="
```

Comment:
Beautiful! Just one suggestion - write the test as `1 = 1` and split by spaces into three components to compare with.